Scripts/RemediationScripts/Remediate-EnableAADAuthentication.ps1

```diff
@@ -0,0 +1,708 @@
+            }
+        }
+        catch
+        {
+            $SQLServersWithADAdminDisabled += $sqlServerInstance 
+            Write-Host "Error fetching Azure AD admin configuration: Resource Group Name - [$($_.ResourceGroupName)], Resource Name - [$($_.ServerName)]. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+        }
+    }
+
+    $totalSQLServersWithADAdminDisabled = ($SQLServersWithADAdminDisabled | Measure-Object).Count
+
+    if ($totalSQLServersWithADAdminDisabled -eq 0)
+    {
+        Write-Host "`nNo SQL Server found with Azure AD admin disabled. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        break
+    }
+
+    Write-Host "`nFound $($totalSQLServersWithADAdminDisabled) SQL Server(s) with Azure AD admin disabled." -ForegroundColor $([Constants]::MessageType.Update)
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\EnableADAdminForSQLServers"
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+
+    # Backing up SQL Server details.
+    $backupFile = "$($backupFolderPath)\SQLServersWithADAdminDisabled.csv"
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 3 of 4] Backing up SQL Server details..."
+    Write-Host "SQL Server(s) details have been backed up to:" -ForegroundColor $([Constants]::MessageType.Update)
+    Write-Host "$($backupFile)"
+
+    $SQLServersWithADAdminDisabled | Export-CSV -Path $backupFile -NoTypeInformation -ErrorAction Stop
+    Write-Host $([Constants]::DoubleDashLine)
+
+    if (-not $DryRun)
+    {
+        Write-Host "[Step 4 of 4] Enabling Azure AD admin for SQL Server(s)..." 
+        Write-Host "Do you want to enable AD Admin for SQL Server(s)?`n" -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+
+        $userInput = Read-Host -Prompt "(Y|N)"
+
+        if ($userInput -ne "Y")
+        {
+            Write-Host "Azure AD admin will not be enabled for any SQL Server. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
+            break
+        }
+
+        # To hold results from the remediation.
+        $remediatedSqlServers = @()
+
+        # Includes SQL Servers that were skipped during remediation. There were errors remediating them.
+        $skippedSqlServers = @()
+
+        $SQLServersWithADAdminDisabled  | ForEach-Object {
+            Write-Host "Checking SQL Server Resource ID: $($_.ResourceId)" 
+            $sqlServerInstance = $_
+            try
+            {
+                if($_.IsSynapseWorkspace -eq $true)
+                {
+                    $adAdmin = Set-AzSynapseSqlActiveDirectoryAdministrator -ResourceGroupName $_.ResourceGroupName -WorkspaceName $_.ServerName -DisplayName $_.DisplayName
```

Comment:
DisplayName coming as user input we should check for null, before attempting to set that as AAD admin.